### DISCUSSION TITLE
Why is the x-nextjs-stale-time always 4294967294?

### DISCUSSION THREAD

**Discussion by yandodov** [vercel/next.js#72398] [Help] [answered]
### Summary

I'm debugging caching and I noticed the following in `/app/.next/server/app/page/example.meta` (the ISR cache):

```json
{
  "headers": {
    "x-nextjs-stale-time": "4294967294",
    "x-next-cache-tags": "..."
  },
  "status": 200
}
```

I can see that `x-nextjs-stale-time` is `4294967294` both on my 404 pages (which are cached for 1 year by default) *and* my regular 200 pages that are cached for 180 seconds.

This doesn't seem to be any timestamp. I thought it maybe measured seconds, but the math doesn't check out:

```
4294967294/60/60/24/365 = 136.1925194698 years??
```

Can someone point out what this number is? Extra credit if someone tells me how to make the `not-found.ts` **not** be 1 year...

**Comment by icyJoseph** [comment]
Interesting... 4294967294 is `Math.pow(2, 32) - 2`, which is the maximum index of a JavaScript array... 

People use that to pull off some crazy tricks, https://github.com/developit/mitt/blob/main/src/index.ts#L86 ~ though this uses 4294967295, which is the maximum length.

Anyhow, now it is too interesting to just ignore that value!

**Reply by icyJoseph** [reply, marked as answer]
So, I think, it comes from here:

- https://github.com/vercel/next.js/blob/1c22d142e09c71d6fd88c72e076a31880d09e0b7/packages/next/src/lib/constants.ts#L38C1-L41C41

```js
// in seconds, represents revalidate=false. I.e. never revaliate.
// We use this value since it can be represented as a V8 SMI for optimal performance.
// It can also be serialized as JSON if it ever leaks accidentally as an actual value.
export const INFINITE_CACHE = 0xfffffffe
```

Which is indeed, `4294967294` ~
